ops/internal/deployer/state.go

```diff
@@ -27,24 +34,66 @@ var standardV3State []byte
 //go:embed configs/v3-intent.toml
 var standardV3Intent []byte
 
-type OpaqueMapping map[string]any
-
-func ReadOpaqueMappingFile(p string) (OpaqueMapping, error) {
+func ReadOpaqueStateFile(p string) (OpaqueState, error) {
 	f, err := os.Open(p)
 	if err != nil {
 		return nil, fmt.Errorf("failed to open JSON file: %w", err)
 	}
 	defer f.Close()
 
-	var out OpaqueMapping
+	var out OpaqueState
 	if err := json.NewDecoder(f).Decode(&out); err != nil {
 		return nil, fmt.Errorf("failed to unmarshal JSON: %w", err)
 	}
 	return out, nil
 }
 
-func MergeStateV2(userState OpaqueMapping) (OpaqueMapping, OpaqueMapping, error) {
-	l1ChainID, err := readL1ChainID(dasel.New(userState))
+func getMergeStateFunc(version string) (func(state OpaqueState) (OpaqueMap, OpaqueState, error), error) {
```

Comment:
Still need the statemerger type alias here.

Reply:
Done in https://github.com/ethereum-optimism/superchain-registry/pull/1041/commits/1fba7e052bf9101c629c66b62fe5ee4e66b0c02d